randovania/games/am2r/exporter/patch_data_factory.py

```diff
@@ -177,15 +177,41 @@ def _create_pickups_dict(
         return pickup_map_dict
 
     def _create_room_dict(self) -> dict:
-        return {
-            area.extra["map_name"]: {
-                "display_name": area.name,
-                "region_name": region.name,
-                "minimap_data": area.extra["minimap_data"],
-            }
-            for region in self.game.region_list.regions
-            for area in region.areas
-        }
+        rng = Random(self.description.get_seed_for_player(self.players_config.player_index))
+
+        return_dict = {}
+        for region in self.game.region_list.regions:
+            for area in region.areas:
+                light_level = None
+                if rng.random() < (self.configuration.darkness_chance / 1000.0):
```

Comment:
I feel 1000 units is more precision than we need and inflates the permalink unecessarily.
But I also feel we're past the point we care much about permalink size 😅

Reply:
Could've used a float, but I dont think it really matters in terms of permalink size.